samza-api/src/main/java/org/apache/samza/context/ApplicationDefinedTaskContext.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.samza.context;
+
+/**
+ * An application should implement this to contain any runtime objects required to initialize the processing logic.
+ * <p>
+ * An instance of this will be created for each task.
+ * <p>
+ * If it is necessary to have a separate instance per task, then use {@link ApplicationDefinedContainerContext} instead.
+ * An {@link ApplicationDefinedTaskContextFactory} needs to be implemented to build instances of this context.
+ */
+public interface ApplicationDefinedTaskContext {
```

Comment:
Does it makes sense to do the following renames. Rename `SamzaContainerContext` to `PredefinedContainerContext` and `SamzaTaskContext` to `PredefinedTaskContext`. Rename `ApplicationDefinedTaskContext` to  `UserDefinedTaskContext` or just `TaskContext` .